cmd/timescaledb-parallel-copy/main.go

```diff
@@ -110,6 +117,19 @@ func main() {
 		csvcopy.WithVerbose(verbose),
 	}
 
+	batchErrorHandler := csvcopy.BatchHandlerError()
+	if skipBatchErrors {
+		batchErrorHandler = csvcopy.BatchHandlerNoop()
+	}
+	if batchErrorOutputDir != "" {
+		log.Printf("batch errors will be stored at %s", batchErrorOutputDir)
+		batchErrorHandler = csvcopy.BatchHandlerSaveToFile(batchErrorOutputDir, batchErrorHandler)
+	}
+	if verbose || skipBatchErrors {
+		batchErrorHandler = csvcopy.BatchHandlerLog(logger, batchErrorHandler)
+	}
+	opts = append(opts, csvcopy.WithBatchErrorHandler(batchErrorHandler))
```

Comment:
So far I think main is the best place to put this logic, moving this to a function won't make it much easier. as you need to pass all the flags.

Reply:
you can have more functions directly in the main.go file so you would still have access to the flags since they're global. It's just a way to avoid an ever-growing main function that happens as programs become more complex